dm/pkg/terror/terror.go

```diff
@@ -204,15 +204,21 @@ func (e *Error) Workaround() string {
 	return e.workaround
 }
 
-// Error implements error interface.
-func (e *Error) Error() string {
+// ErrorWithoutWorkaround returns err msg without workaround, in some place like cloud we want display it separately.
```

Comment:
I remember the `instruction` fields have been filled for all the check items (i.e., `Results[i].Instruction`). Where can we use this function?

Reply:
if precheck fail with err, there is no instruction. it's used in `markCheckError`

Reply:
https://github.com/pingcap/tiflow/blob/a2d2a5213c3963485e5cb9f3bd7b6967302b9bb6/dm/pkg/checker/privilege.go#L110-L114

The `instruction` is set in the `Result` and `dm-on-cloud` uses this field as the `solution`. Do you mean it's only used in op?